src/symbols_linux.cpp

```diff
@@ -775,6 +775,19 @@ void Symbols::parseLibraries(CodeCacheArray* array, bool kernel_symbols) {
     std::unordered_map<u64, SharedLibrary> libs;
     collectSharedLibraries(libs, MAX_NATIVE_LIBS - array->count());
 
+    char* exePath = realpath("/proc/self/exe", NULL);
+    if (exePath == NULL) {
+        char buf[PATH_MAX];
+
+        // realpath() may fail for a path like /proc/[pid]/root/bin/asprof
+        // In this case, resolve the link as is.
+        ssize_t size = readlink("/proc/self/exe", buf, sizeof(buf) - 1);
+        if (size >= 0) {
+            buf[size] = 0;
+            exePath = buf;
+        }
+    }
```

Comment:
There is a simpler and more efficient way to find main executable:
```
    const void* main_phdr = NULL;
    dl_iterate_phdr([](struct dl_phdr_info* info, size_t size, void* data) {
        *(const void**)data = info->dlpi_phdr;
        return 1;
    }, &main_phdr);
```
Then check if `main_phdr >= lib.image_base && main_phdr < lib.map_end`

Calling `dl_iterate_phdr` should be fine as long as there are no allocations inside.